Feature/issue 1 typescript UI rewrite

mads5606 opened the pull request.

mads5606 committed:
Rewritting header to typescrip

Note: Some div elements throw jsx errors: "JSX element implicitly has type 'any' because no interface 'JSX.IntrinsicElements' exists"
Seems to be a react error we can ignore. Will look further into it.

Implemented:
 - rewrote the header to typescript
 - Changed import to reflect default import

mads5606 committed:
Import changes

mads5606 committed:
Finished header

Notes: Tested that it actually works.

Implemented
 - Last changes for header to be in typescript and working.
 - Fixed error-message from previous commit, but it was largely irrelevant.

mads5606 committed:
Added the footer

Notes: Footer should still be changed to tsx.

Implemented:
 - Added the actual footer to the ui.
 - Changed to footer link.

mads5606 committed:
Rewrites and prop types

Notes: Mostly just fiddling with types for typescript. Some parts left with 'any' types.

Implemented
 - ContainerTable, NewContainerDialog, RenameContainerDialog and footer componenent to typescript
 - Added null type for the ressource page
 - Renaming some style files

mads5606 committed:
More types and rewrites

Implemented:
 - Notification_menu and Notifier componenents changed to TypeScript.
 - Replaced more of the 'any' types so we actually use what TypeScript gives us.

mads5606 committed:
Finishing Typescript Rewrite

Notes: Needs more testing and some types could be changed a bit.
There is an issue with the use of withStyles in the NotificationMenu Component, freezez the page when
in typescript. Will look into it more, later.

Implemented
 - InspectContainer, ContainerTable, NewContainer and ReconfigureContainer components rewritten.
 - CPU, Harddisk, Memory, Policy section Components rewritten.
 - RenameContainerDialog rewritten.
 - Tightened up some types, replaced some 'any' types.

mads5606 committed:
Fixed issues with withStyles and typescript

Note: Turns out that if you read the documentation properly your life is a lot easiere, who would have guessed?
Just some small changes needed that were documentated in materialUI, but I missed at first.

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -90,14 +120,22 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
             <Grid container spacing={2}>
-              <Grid item xs={12} style={{ padding: "2px", paddingLeft: "12px" }}>
-                <Typography variant="subtitle2">Server to Start Container On</Typography>
+              <Grid
+                item
+                xs={12}
+                style={{ padding: "2px", paddingLeft: "12px" }}
+              >
```
I'm not sure that I agree that this has to be exploded into more lines.

mads5606 replied:
Agreed, no need to.

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -76,9 +104,11 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
+      return;
     }
-    var name: any = Object.keys(templates).find((template) => template === templateName);
+    var name: any = Object.keys(templates).find(
+      (template) => template === templateName
+    );
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -52,14 +72,22 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
     console.log(JSON.stringify(valuesToSend));
     if (socketConnection !== undefined)
-      socketConnection.invoke(CREATE_NEW_CONTAINER_REQUEST, selectedServer, JSON.stringify(valuesToSend));
+      socketConnection.invoke(
+        CREATE_NEW_CONTAINER_REQUEST,
+        selectedServer,
+        JSON.stringify(valuesToSend)
+      );
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -33,9 +49,13 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
     networkMode: "",
   });
-  const [selectedServer, setSelectedServer] = React.useState<string>(servers[0].url);
+  const [selectedServer, setSelectedServer] = React.useState<string>(
+    servers[0].url
+  );
```
Unnecessary?

mads5606 replied:
Agreed.

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -33,9 +49,13 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
     volumesFrom: [],
     networkMode: "",
   });
-  const [selectedServer, setSelectedServer] = React.useState<string>(servers[0].url);
+  const [selectedServer, setSelectedServer] = React.useState<string>(
+    servers[0].url
+  );
   const [selectedTemplate, setSelectedTemplate] = React.useState<string>("");
-  const socketConnection: HubConnection | undefined = useSelector((store: IRootState) => store.containerData.socketConnection);
+  const socketConnection: HubConnection | undefined = useSelector(
+    (store: IRootState) => store.containerData.socketConnection
+  );
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -120,8 +160,14 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
               </Grid>
-              <Grid item xs={12} style={{ padding: "2px", paddingLeft: "12px" }}>
-                <Typography variant="subtitle2">Choose a Template Here</Typography>
+              <Grid
+                item
+                xs={12}
+                style={{ padding: "2px", paddingLeft: "12px" }}
+              >
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -135,7 +181,9 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                   fullWidth
                   value={selectedTemplate}
-                  onChange={(event: any) => handleAutoFillTemplate(event.target.value)}
+                  onChange={(event: any) =>
+                    handleAutoFillTemplate(event.target.value)
+                  }
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -145,7 +193,11 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                 </TextField>
               </Grid>
-              <Grid item xs={12} style={{ padding: "2px", paddingLeft: "12px" }}>
+              <Grid
+                item
+                xs={12}
+                style={{ padding: "2px", paddingLeft: "12px" }}
+              >
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -224,7 +280,11 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
 
               <Grid container alignItems="center" justify="center">
-                <Grid item xs={10} style={{ padding: "2px", paddingLeft: "12px" }}>
+                <Grid
+                  item
+                  xs={10}
+                  style={{ padding: "2px", paddingLeft: "12px" }}
+                >
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -239,7 +299,10 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                       }}
                     >
-                      <RemoveIcon color="primary" aria-label="delete port binding" />
+                      <RemoveIcon
+                        color="primary"
+                        aria-label="delete port binding"
+                      />
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -249,7 +312,10 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                         setValues({
                           ...values,
-                          ports: [...values.ports, { containerPort: "", hostPort: "" }],
+                          ports: [
+                            ...values.ports,
+                            { containerPort: "", hostPort: "" },
+                          ],
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -307,8 +373,14 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
               <Grid container alignItems="center" justify="center">
-                <Grid item xs={10} style={{ padding: "2px", paddingLeft: "12px" }}>
-                  <Typography variant="subtitle2">Environment Variables</Typography>
+                <Grid
+                  item
+                  xs={10}
+                  style={{ padding: "2px", paddingLeft: "12px" }}
+                >
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -322,7 +394,10 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                       }}
                     >
-                      <RemoveIcon color="primary" aria-label="delete environment variable" />
+                      <RemoveIcon
+                        color="primary"
+                        aria-label="delete environment variable"
+                      />
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -332,11 +407,17 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                         setValues({
                           ...values,
-                          environment: [...values.environment, { key: "", value: "" }],
+                          environment: [
+                            ...values.environment,
+                            { key: "", value: "" },
+                          ],
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -332,11 +407,17 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                       }}
                     >
-                      <AddIcon color="primary" aria-label="add environment variable" />
+                      <AddIcon
+                        color="primary"
+                        aria-label="add environment variable"
+                      />
```
Unnecessary?

mads5606 replied:
Agreed

dstoft commented on interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx:
```diff
@@ -388,7 +469,11 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
 
               <Grid container alignItems="center" justify="center">
-                <Grid item xs={10} style={{ padding: "2px", paddingLeft: "12px" }}>
+                <Grid
+                  item
+                  xs={10}
+                  style={{ padding: "2px", paddingLeft: "12px" }}
+                >
```
Unnecessary?

mads5606 replied:
Agreed

dstoft reviewed (changes requested):
I really like the changes, it will make it way easier to maintain.
Except for a few lines in "NewContainerDialog.component.tsx", you have exploded a few lines into several lines, where I disagree on the readability improvement. I believe it may have been caused by your IDE attempting to enforce a maximum line length.
Normally I would not comment on this, but since this PR will probably decide how our code looks in the future, we should probably address it.
What is your opinion on this? 

Such as:
> (OLD) const [selectedServer, setSelectedServer] = React.useState<string>(servers[0].url);
> (NEW) const [selectedServer, setSelectedServer] = React.useState<string>(
> (NEW) servers[0].url
> (NEW) );

mads5606 commented:
> I really like the changes, it will make it way easier to maintain.
> Except for a few lines in "NewContainerDialog.component.tsx", you have exploded a few lines into several lines, where I disagree on the readability improvement. I believe it may have been caused by your IDE attempting to enforce a maximum line length.
> Normally I would not comment on this, but since this PR will probably decide how our code looks in the future, we should probably address it.
> What is your opinion on this?
> 
> Such as:
> 
> > (OLD) const [selectedServer, setSelectedServer] = React.useState(servers[0].url);
> > (NEW) const [selectedServer, setSelectedServer] = React.useState(
> > (NEW) servers[0].url
> > (NEW) );

I completely agree, the IDE is a bit over-zealous. Much prefer keeping it single-line when possible, especially when its a single expression, and not an encapsulation of several statements. I have changed the formatting to reflect it.

mads5606 committed:
Updated formatting

Changed the formatting with respect to feedback
on the review. Generally avoiding newlines on
single expression statements.
